pkg/server/v2/server.go

```diff
@@ -72,6 +72,66 @@ type Callbacks interface {
 	OnFetchResponse(*discovery.DiscoveryRequest, *discovery.DiscoveryResponse)
 }
 
+// CallbackFuncs is a convenience type for implementing the Callbacks interface.
+type CallbackFuncs struct {
```

Comment:
What are your thoughts on adding a test that uses this convenience struct?

Reply:
Yeh I didn't come up with a reasonable way to test this. The test pkg needs a struct with state and it seemed pretty ugly to hack the callback funds over the top of `test.Callbacks`. Happy to add tests if you have any ideas on how to best approach it?

Reply:
What about something like this?
```Go
diff --git pkg/test/main/main.go pkg/test/main/main.go
index 2e75619..566ac03 100644
--- pkg/test/main/main.go
+++ pkg/test/main/main.go
@@ -95,8 +95,25 @@ func main() {

        configv2 := cachev2.NewSnapshotCache(mode == resourcev2.Ads, cachev2.IDHash{}, logger{})
        configv3 := cachev3.NewSnapshotCache(mode == resourcev2.Ads, cachev3.IDHash{}, logger{})
-       srv2 := serverv2.NewServer(context.Background(), configv2, cbv2)
-       srv3 := serverv3.NewServer(context.Background(), configv3, cbv3)
+
+       srv2 := serverv2.NewServer(context.Background(), configv2, serverv2.CallbackFuncs{
+               StreamOpenFunc:     cbv2.OnStreamOpen,
+               StreamClosedFunc:   cbv2.OnStreamClosed,
+               StreamRequestFunc:  cbv2.OnStreamRequest,
+               StreamResponseFunc: cbv2.OnStreamResponse,
+               FetchRequestFunc:   cbv2.OnFetchRequest,
+               FetchResponseFunc:  cbv2.OnFetchResponse,
+       })
+
+       srv3 := serverv3.NewServer(context.Background(), configv3, serverv3.CallbackFuncs{
+               StreamOpenFunc:     cbv3.OnStreamOpen,
+               StreamClosedFunc:   cbv3.OnStreamClosed,
+               StreamRequestFunc:  cbv3.OnStreamRequest,
+               StreamResponseFunc: cbv3.OnStreamResponse,
+               FetchRequestFunc:   cbv3.OnFetchRequest,
+               FetchResponseFunc:  cbv3.OnFetchResponse,
+       })
+
        alsv2 := &testv2.AccessLogService{}
        alsv3 := &testv3.AccessLogService{}
```

Reply:
> How about changing the code in server_test to use this struct? 

Great idea, done!